worker/src/RTC/SimulcastConsumer.cpp

```diff
@@ -691,7 +692,7 @@ namespace RTC
 		bool isSyncPacket = this->syncRequired;
 
 		// Sync sequence number and timestamp if required.
-		if (isSyncPacket)
+		if (isSyncPacket && (this->spatialLayerToSync == -1 || this->spatialLayerToSync == spatialLayer))
```

Comment:
Why should we accept such a -1 condition?

Reply:
Yes, good question. We better not, I suppose. However, as far as I understand when we set `syncRequired` true in `UserOnTransportConnected` and `UserOnResumed` we don't know which layer is going to be set as current yet. We may know the target layer after `MayChangeLayers` gets executed, though. But in that case, it seems that it doesn't really make sense to set `syncRequired` in either of these methods because it will be set later in `SendRtpPacket`.

Are there a scenario in which either `currentSpatialLayer` or `targetSpatialLayer` do not equal `-1` before `UserOnResumed` or `UserOnTransportConnected` get called?

Reply:
`this->spatialLayerToSync == -1` is not needed here since `syncRequired` (which is assigned to `isSyncPacket`) and `spatialLayerToSync` are always set at the same time.

Reply:
syncRequired must be set in `UserOnResumed` otherwise we would forward any RTP packet on resume, which we don't want.

Reply:
@dsdolzhenko, the condition is correct. Let's keep the `this->spatialLayerToSync == -1` because it's needed in case we come from resuming the Consumer. In which case `spatialLayerToSync` is unset (-1)